llmfoundry/eval/datasets/in_context_learning_evaluation.py

```diff
@@ -134,6 +136,8 @@ def __init__(
         static_keys: Optional[List] = None,
         list_keys: Optional[List] = None,
         tensor_keys: Optional[List] = None,
+        *args: Any,
```

Comment:
I think that if you adjust `build_icl_dataloader` to take in arbitrary kwargs and then pass those along directly to the registry constructor, you don't need to add arbitrary support for kwargs here, right?

Reply:
I think this will require a couple other changes too. I think we should push all of the config defaults to the actual class constructors instead of having them in `build_icl_dataloader` and `get_icl_task_dataloader`. If `get_icl_task_dataloader` just takes arbitrary kwargs containing the optional args, then I believe you should be able to not have any special casing for different class types. Required args for all ICL classes can be named, and optional args that only some subclasses have can be just kwargs that get passed along. Let me know what you think!

Reply:
Yes I agree. The `super().__init__` calls in every specific ICL Dataset's implementation will become verbose, but I think that is a better design than not keeping the fields as named arguments.